src/runner/index.js

```diff
@@ -126,6 +137,10 @@ export default class Runner extends EventEmitter {
                 return this._runTask(reporterPlugin, browserSet, tests);
             });
 
-        return Runner._createCancelablePromise(runTaskPromise);
+        return this._createCancelablePromise(runTaskPromise);
+    }
+
+    async stop () {
+        await Promise.all(this.pendingTaskPromises.slice().map(taskPromise => taskPromise.cancel()));
```

Comment:
Unfortunately, I have realized that `for..of` doesn't iterate over an element that is next to deleted one. So when trying to do
```
var a = [1,2,3,4,5,6]; 
for(var x of a) 
    a.splice(a,indexOf(x), 1);
```
`a` will be equal to `[2,4,6]`.